cmake/EthCompilerSettings.cmake

```
@@ -54,9 +54,10 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	add_compile_options(-Wsign-conversion)
 	add_compile_options(-Wconversion)
 
-	eth_add_cxx_compiler_flag_if_supported(
-		$<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>
-	)
+	check_cxx_compiler_flag(-Wextra-semi WEXTRA_SEMI)
+	if(WEXTRA_SEMI)
+		add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>)
+	endif()
```

Can you check if this would work?
```suggestion
    if ($<COMPILE_LANGUAGE:CXX>)
        eth_add_cxx_compiler_flag_if_supported(-Wextra-semi)
    endif()
```

---

This will not work. The `if` is evaluated at "configure" time, the generator expressions are evaluated at "generation" time. How this case works in practice is: at generation time the compile flags are applied to each source file and then it is known what language the file is. During configuration you at most has "enabled" languages (C and C++ in solidity case probably).

---

Oh, thanks, now I see where the problem is.

In that case how about adding an optional argument to `eth_add_cxx_compiler_flag_if_supported()` to specify the language and moving the generator expression inside?